### PR TITLE
Optimize URL rewrite indexes

### PR DESCRIPTION
**Pull request by toddbc**
With the optimizations that make category/product reindexing fast, URL rewrites are now near the top of full reindex time on some sites.

These changes improve partial and full reindexing for both URL redirects and category URL rewrites.  Obviously makes the biggest difference when you have a large number, as many large Magento sites do.

### CODE REVIEW

File: app/code/community/SomethingDigital/EnterpriseIndexPerf/Model/Url/Rewrite/Redirect/Refresh.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    /**
+     * Clean old url rewrites records from table
+     *
+     * @return Enterprise_UrlRewrite_Model_Index_Action_Url_Rewrite_RefreshAbstract
+     */
+    protected function _cleanOldUrlRewrite()
+    {
+        $selects = $this->_prepareSelectsByRange($this->_getCleanOldUrlRewriteSelect(), 'url_rewrite_id');
+        foreach ($selects as $select) {
+            $this->_connection->query($select->deleteFromSelect('ur'));
```
Comment (philwinkle) [inline app/code/community/SomethingDigital/EnterpriseIndexPerf/Model/Url/Rewrite/Redirect/Refresh.php:17]: `ur` is correct, I assume.

**Comment by philwinkle** [comment]
LGTM but needs rebase (sorry)

File: app/code/community/SomethingDigital/EnterpriseIndexPerf/Model/Url/Rewrite/Redirect/Refresh.php (hunk 1 above)
Comment (toddbc) [inline app/code/community/SomethingDigital/EnterpriseIndexPerf/Model/Url/Rewrite/Redirect/Refresh.php:17]: Yes, that's the same alias it deletes from in the base code - just without the chunking.

**Comment by toddbc** [comment]
Rebased - yeah, the XML changes were too close together.

**Pull request merged**